cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/BrokerStateProvider.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.resource;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.operator.cluster.model.DnsNameGenerator;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.Util;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.PemTrustOptions;
+
+public class BrokerStateProvider {
+
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(BrokerStateProvider.class.getName());
+
+    protected String namespace;
+    private static final String BROKER_STATE_REST_PATH = "/v1/broker-state";
+    private static final int BROKER_STATE_HTTPS_PORT = 8443;
+    private static final String CLUSTER_CA_CERT_SECRET_KEY = "ca.crt";
+    private static final String CO_KEY_SECRET_KEY = "cluster-operator.key";
+    private static final String CO_KEY_CERT_SECRET_KEY = "cluster-operator.crt";
+
+    private Vertx vertx;
+    private Reconciliation reconciliation;
+    private String cluster;
+    private PemTrustOptions pto;
+    private PemKeyCertOptions pkco;
+
+    BrokerStateProvider(Reconciliation reconciliation, Vertx vertx, String cluster, String namespace, Secret clusterCaCertSecret, Secret coKeySecret) {
```

Comment:
Given that we're trying to move away from using Vertx I wonder if we should use [`java.net.http.HttpClient`](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpClient.html). `doGet()` would then be blocking, returning a `String`, rather than a `Future<String>`, but that seems fine since you're already doing a blocking get on line 112, so you're not really benefitting from the fact that it returns a future anyway. Wdyt @ppatierno @scholzj?

Reply:
I agree, yes.